omnigan/trainer.py

```diff
@@ -1569,17 +1613,21 @@ def masker_s_loss(self, x, z, target, domain, for_="G"):
                         full_loss += gp_loss
                     else:
                         raise NotImplementedError
-
+        if self.opts.gen.m.use_spade:
+            return full_loss, softmax(pred, dim=1).detach()
```

Comment:
@tianyu-z I think it would be clearer if we kept a constant API: `masker_X_loss` should always return a tuple `loss, pred` it's just that if we don't use spade for `m` then we'll just ignore the second item of the tuple